crates/sui-indexer-alt-jsonrpc/src/data/object_versions.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeSet, HashMap},
+    sync::Arc,
+};
+
+use async_graphql::dataloader::Loader;
+use diesel::{ExpressionMethods, QueryDsl};
+use sui_indexer_alt_schema::{objects::StoredObjVersion, schema::obj_versions};
+use sui_types::base_types::ObjectID;
+
+use super::reader::{ReadError, Reader};
+
+/// Key for fetching the latest version of an object, not accounting for deletions or wraps. If the
+/// object has been deleted or wrapped, the version before the delete/wrap is returned.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub(crate) struct LatestObjectKey(pub ObjectID);
```

Comment:
When I first saw this key, I thought it would fetch me the latest object because of its name, but it actually only fetches the latest version. I don't know what the convention is on key naming and it's unfortunate that we can't have `Loader` parametrized by both key and value (or can we?). Is there a name that more clearly shows that  this loads the object version?

Reply:
>  I don't know what the convention is on key naming and it's unfortunate that we can't have Loader parametrized by both key and value (or can we?). Is there a name that more clearly shows that this loads the object version?

We can call it `LatestObjectVersionKey`? You're right that the loader is only parameterized by its key, which is why the convention is that we introduce wrapper types for keys that indicate the value that the key will produce (This is one of the issues with the current basic object support -- it uses a bare tuple as the loader key -- this is not a good idea, because as you have identified, it doesn't really indicate what data you get when you load that key).

Reply:
I ended up making this change at the top of the PR stack, to avoid having to pipe it through the code moves between here and there.